pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
 #  publicly, and to permit other to do so.
 #  ___________________________________________________________________________
 
+import itertools
 
-from pyomo.common.dependencies import (
-    numpy as np, numpy_available,
-    pandas as pd, pandas_available,
-    matplotlib as plt, matplotlib_available,
-)
+class SpecialSet: 
+    def __init__(self):
+        """This class defines variable names with provided names and indexes.
+        """
+        self.special_set = []
 
+    def specify(self, self_define_res):
```

Comment:
Changed. I don't think this function needs any error checking as long as the string name can be found as a variable in the Pyomo model given, which is an error checking in `doe` not here (because here we don't know the model yet.)